Integrations/integration-ATD.yml

```diff
@@ -0,0 +1,545 @@
+description: 'Integrated advanced threat detection: Enhancing protection from network
+  edge to endpoint'
+detaileddescription: |
+  When entering a is a user to use the API with, please make sure the user has capability of "Allow Multiple Logins".
+configuration:
+- display: Server URL (e.g. https://192.168.0.1)
+  name: baseUrl
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: username
+  name: username
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: password
+  name: password
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: "true"
+  type: 8
+  required: false
+script:
+  script: |2-
+
+    var base = params.baseUrl;
+
+    var Base64 = {
+      _keyStr: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=",
+      encode: function(input) {
+        var output = "";
+        var chr1, chr2, chr3, enc1, enc2, enc3, enc4;
+        var i = 0;
+        while (i < input.length) {
+          chr1 = input.charCodeAt(i++);
+          chr2 = input.charCodeAt(i++);
+          chr3 = input.charCodeAt(i++);
+          enc1 = chr1 >> 2;
+          enc2 = ((chr1 & 3) << 4) | (chr2 >> 4);
+          enc3 = ((chr2 & 15) << 2) | (chr3 >> 6);
+          enc4 = chr3 & 63;
+          if (isNaN(chr2)) {
+            enc3 = enc4 = 64;
+          } else if (isNaN(chr3)) {
+            enc4 = 64;
+          }
+          output = output + this._keyStr.charAt(enc1) + this._keyStr.charAt(enc2) + this._keyStr.charAt(enc3) + this._keyStr.charAt(enc4);
+        }
+
+        return output;
+      }
+    };
+
+    var getSession = function() {
+        res = sendRequest('/php/session.php','GET', loginHeaders);
+        if (res) {
+            return res.results;
+        }
+    };
+
+
+    // handle '/' at the end of the url
+    if (base[base.length - 1] === '/') {
+        base = base.substring(0, base.length - 1);
+    }
+
+    var username = params.username;
+    var password = params.password;
+    var insecure = params.insecure;
+
+    var loginHeaders = {'Accept': ['application/vnd.ve.v1.0+json'], 'Content-Type': ['application/json'], 'VE-SDK-API':[Base64.encode(username + ':' + password)]};
```

Comment:
maybe it's better to change the server so it would push basic auth to other places